server/py/framework/db/sqldb/db.py

```diff
@@ -7143,3 +7146,101 @@ def _paginate_query(
             query = query.limit(limit)
 
         return query
+
+    @staticmethod
+    def _find_artifact_latest_tag(session, object_record):
+        """Find the 'latest' tag for an artifact object."""
+        return (
+            session.query(ArtifactV2.Tag)
+            .filter(
+                ArtifactV2.Tag.obj_id == object_record.id,
+                ArtifactV2.Tag.name == mlrun.common.constants.RESERVED_TAG_NAME_LATEST,
+                ArtifactV2.Tag.project == object_record.project,
+            )
+            .one_or_none()
+        )
+
+    @staticmethod
+    def _find_other_latest_in_other_iterations(session, object_record):
+        """Check if there are other 'latest' tags in other iterations for the same producer_id."""
+        return (
+            session.query(ArtifactV2.Tag)
+            .filter(
+                ArtifactV2.Tag.obj_id != object_record.id,
+                ArtifactV2.Tag.name == mlrun.common.constants.RESERVED_TAG_NAME_LATEST,
+                ArtifactV2.Tag.project == object_record.project,
+                ArtifactV2.Tag.obj_id.in_(
+                    session.query(ArtifactV2.id).filter(
+                        ArtifactV2.producer_id == object_record.producer_id,
+                        ArtifactV2.iteration != object_record.iteration,
+                    )
+                ),
+            )
+            .first()
```

Comment:
`.all()`?

Reply:
@rokatyy we are using `first()` here because we are interested in checking whether any matching tags exist. When using `first()`, the search stops once the first match is found.

Reply:
@moranbental rename it then, because the name is confusing